components/event-bus/internal/knative/subscription/controller/subscription/reconcile.go

```diff
@@ -205,11 +199,11 @@ func (r *reconciler) reconcile(ctx context.Context, subscription *eventingv1alph
 	return false, nil
 }
 
-func (r *reconciler) deleteExternalDependency(ctx context.Context, subscription *eventingv1alpha1.Subscription, channelName string, channelNamespace string) error {
+func (r *reconciler) deleteExternalDependency(ctx context.Context, knativeSubsName string, knativeSubsURI string, channelName string, namespace string) error {
 	log.Info("Deleting the external dependencies")
 
 	// In case Knative Subscription exists, delete it.
-	knativeSubs, err := r.knativeLib.GetSubscription(subscription.Name, channelNamespace)
+	knativeSubs, err := r.knativeLib.GetSubscription(knativeSubsName, namespace)
 	if err != nil && !errors.IsNotFound(err) {
 		return err
 	} else if err == nil && knativeSubs != nil {
```

Comment:
Is the case where the `knativeSubs == nil` handled?

Reply:
If "err == nil" then knativeSubs exists and it's always "!= nil".
so that the additional check "knativeSubs != nil" can be removed.

Reply:
if Knative Subscription is nil, we don't need to do anything. There is nothing to delete.